### DISCUSSION TITLE
Pushing to `content` in consumer `tailwind.config.js` for Component Library

### DISCUSSION THREAD

**Discussion by Ademsk1** [tailwindlabs/tailwindcss#12491] [Help] [answered]
Hi folks. 

I'm writing a component library that is written in JSX and tailwind. As the library was written in Tailwind, I need to ask the consumer of the library to add `./node_modules/my-package/**/*,js` to their `content` array in their tailwind config. Is there a way to avoid this?

I was thinking of writing a plugin / preset, but access to the `config()` from the `plugin(function({config})..)` seems purely for read-only purposes, and if using `preset` it seems my content is overwritten by the users `content`. But if this were possible, a consumer would merely have to add the 
`plugins: [require('my-package')]`

Any thoughts on how I could go about this? I'm not confident as I see a lot of component libraries using Tailwind also require their package to be present in the consumers content.

**Comment by wongjn** [comment]
You could export a value or function from your package that the consumers should use to add your package's path(s) to their `content` array.

**Comment by Ademsk1** [comment, marked as answer]
Turns out the `config` passed into the `plugin` function isn't a deep copy, but the actual consuming applications config (passed by reference), and therefore modifiable. I was able to do the following:
```ts
// my-package/plugin.ts
const myPlugin = (node_modules_path?: string) => {
  return plugin(
    ({ config }: any) => {
      const twConfig = config()
      let myPkgLoc = './node_modules/my-package'
      myPkgLoc = path.join(
        node_modules_path ?? '',
        './node_modules/my-package'
      )

      if (!twConfig.content.files.includes(myPkgLoc)) {
        twConfig.content.files.push(myPkgLoc)
      }
    },
    {
      theme: {
        ...myconfig.theme
      }
    }
  )
}

module.exports = myPlugin

```

```ts
// consuming-app/tailwind.config.js
module.export = {
  ...
  plugins: [require('my-package')()]
}
```
It also allows for relative paths to be used if that's enabled in the content section.
